### PR TITLE
pca9685: cleanup

### PR DESCRIPTION
**Pull request by ether42**
The PCA9685's datasheet is available [here](https://cdn-shop.adafruit.com/datasheets/PCA9685.pdf).

First commit cleanups some stuff you commented on in https://github.com/google/periph/pull/296.

Second commit makes use of the auto-increment capability of the device to simplify contiguous writes. You've talked about `mmr`, do you think it's still required?

I tested it with a couple of servos. Do you have plans for the [servo module](https://github.com/google/periph/blob/master/experimental/devices/pca9685/servo.go)? I feel like one should be able to associate some `Servo`s to different `ServoGroup`s and the channel should be abstracted away (for example it could be possible to set the angle of all the `Servo`s in the `ServoGroup`).

### CODE REVIEW

File: experimental/devices/pca9685/pca9685.go [hunk 1]
```diff
@@ -89,7 +88,7 @@ func (d *Dev) init() error {
 	return d.SetPwmFreq(50 * physic.Hertz)
 }
 
-// SetPwmFreq set the pwm frequency
+// SetPwmFreq set the pwm frequency.
```
[the remark this answers is not in the document]
Comment (ether42) [reply experimental/devices/pca9685/pca9685.go]: I know of this one but I was a bit reluctant to change the name of the functions. Experimental doesn't have any compatibility policy, so should we anyway?

**Comment by ether42** [comment]
Squashed commits and added an helper function to avoid redundancy.

File: experimental/devices/pca9685/pca9685.go [hunk 2]
```diff
@@ -111,40 +110,29 @@ func (d *Dev) SetPwmFreq(freqHz physic.Frequency) error {
...
 	time.Sleep(100 * time.Millisecond)
 
-	_, err := d.dev.Write([]byte{mode1, (byte)(oldmode | 0x80)})
+	_, err := d.dev.Write([]byte{mode1, oldmode | restart})
 	return err
 }
 
-// SetAllPwm set a pwm value for all outputs
+// setPWM writes a PWM value in a specific register.
+func (d *Dev) setPwm(register uint8, on, off gpio.Duty) error {
```
Comment (maruel) [inline experimental/devices/pca9685/pca9685.go]: Update the function name too, not just the comment.

Let's punt on the public ones for now to not break clients yet, but the private one can be fixed right away.

Comment (ether42) [reply experimental/devices/pca9685/pca9685.go]: That was an oversight on my part, sorry.

**Comment by maruel** [comment]
gohci

**Comment by maruel** [comment]
Thanks!

**Pull request merged**